src/clients/Device.js

```diff
@@ -811,15 +813,13 @@ class Device extends EventEmitter {
   };
 
   releaseOwnership = (flasher: Flasher) => {
-    logger.log('releasing flash ownership ', { deviceID: this.getDeviceID() });
+    logger.info({ deviceID: this.getDeviceID() }, 'releasing flash ownership ');
     if (this._owningFlasher === flasher) {
       this._owningFlasher = null;
     } else if (this._owningFlasher) {
       logger.error(
-        'cannot releaseOwnership, ',
-        flasher,
-        " isn't the current owner ",
-        { deviceID: this.getDeviceID() },
+        { deviceID: this.getDeviceID(), flasher },
+        'cannot releaseOwnership, isn\'t  current owner'
```

Comment:
What do you MEAN (sorry, but there seems to be a problem or a missing configuration)

Let us find the Problem and not talking about "you didn't do something" or "did you run tests".
I will do my best to deliver a good result, and will try to fix all problems!

### pre-commit (from my site, not only mentioned in package.json!)
 
* npm run lint-staged
* npm run lint
* npm run test
* flow check

Everything was DONE with THIS Result (right now!)

```
$ npm run lint

> spark-protocol@0.1.6 lint D:\keastick\RemoteRep\spark-protocol
> eslint --fix --max-warnings 0 -- .


$ npm run lint-staged

> spark-protocol@0.1.6 lint-staged D:\keastick\RemoteRep\spark-protocol
> lint-staged

 ↓ Running tasks for src/**/*.js [skipped]
   → No staged files match src/**/*.js
 ↓ Running tasks for test/**/*.js [skipped]
   → No staged files match test/**/*.js
 
npm run test

> spark-protocol@0.1.6 test D:\keastick\RemoteRep\spark-protocol
> ava --no-cache


  √ should subscribe to event
  √ should listen for public event from another owner device
  √ should filter private event
  √ should filter internal event
  √ should filter event by connectionID
  √ should filter event by deviceID
  √ should filter broadcasted events
  √ should listen for mydevices events only
  √ should unsubscribe all subscriptions by subsriberID
  √ should publish and listen for response

  10 tests passed [09:34:15]

$ flow check

src/lib/Handshake.js:325
325:         deviceProvidedPem,
             ^^^^^^^^^^^^^^^^^ null. This type is incompatible with the expected param type of
 91:     publicKeyPem: string,
                       ^^^^^^ string. See: src/lib/CryptoManager.js:91

src/lib/Handshake.js:325
325:         deviceProvidedPem,
             ^^^^^^^^^^^^^^^^^ undefined. This type is incompatible with the expected param type of
 91:     publicKeyPem: string,
                       ^^^^^^ string. See: src/lib/CryptoManager.js:91

src/scripts/update-firmware-binaries.js:82
 82:     file.on('finish', (): void => file.close((): void => resolve(filename)));
                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ unused function argument
  728:     close(): void
           ^^^^^^^^^^^^^ function type expects no arguments. See lib: C:\Users\ahr\AppData\Local\Temp\flow\flowlib_c41dfb6\node.js:728


Found 3 errors
```